src/test/java/org/apache/sysds/test/AutomatedTestBase.java

```diff
@@ -978,19 +989,26 @@ protected void runRScript(boolean newWay) {
 			}
 
 			long t1 = System.nanoTime();
-			System.out.println("R is finished (in " + ((double) t1 - t0) / 1000000000 + " sec)");
+
+			LOG.info("R is finished (in " + ((double) t1 - t0) / 1000000000 + " sec)");
 		}
 		catch(Exception e) {
-			e.printStackTrace();
-			StringBuilder errorMessage = new StringBuilder();
-			errorMessage.append("failed to run script " + executionFile);
-			errorMessage.append("\nexception: " + e.toString());
-			errorMessage.append("\nmessage: " + e.getMessage());
-			errorMessage.append("\nstack trace:");
-			for(StackTraceElement ste : e.getStackTrace()) {
-				errorMessage.append("\n>" + ste);
+			if(e.getMessage().contains("ERROR: R has ended irregularly")){
+				StringBuilder errorMessage = new StringBuilder();
+				errorMessage.append(e.getMessage());
+				fail(errorMessage.toString());
+			}else {
+				e.printStackTrace();
+				StringBuilder errorMessage = new StringBuilder();
+				errorMessage.append("failed to run script " + executionFile);
+				errorMessage.append("\nexception: " + e.toString());
+				errorMessage.append("\nmessage: " + e.getMessage());
+				errorMessage.append("\nstack trace:");
+				for(StackTraceElement ste : e.getStackTrace()) {
+					errorMessage.append("\n>" + ste);
+				}
```

Comment:
Why do we use `getStackTraceString` at line 1194, but here we don't?

Reply:
Im not sure if i get this comment, at line 1194 for me is: 
```
errorMessage.append("\nfailed to run script: " + executionFile);
```

If i understand correctly, then you are getting confused about the method name at line 1198

```
errorMessage.append(getStackTraceString(e, 0));
```

This is not javas standard getStackTrace. but a method i added to return a custom string of the stack trace. 
Previously the tests would only print one stack trace layer, resulting in a trace that most of the time was useless, and furthermore it would not give out the actual exception method of the lowest level stack. 
This means in practice that the error message most of the time was at script block level saying error in lines between 31 - 33 instead of index out of bounds at exactly line x in file y.

Reply:
Oh, I might have confused the lines. I understood that this is a method you made yourself, but shouldn't you be able to use the same function, because the segment you replaced below is equal (if I am not mistaken) to the one here.